[BUG] test_parquet_testing_error_files - Failed: DID NOT RAISE <class 'Exception'> in databricks runtime 12.2

**Describe the bug**
found the test failures in recent nightly build, like rapids_databricks_nightly-dev-github build ID: 830
and premerge CI w/ databricks test, like rapids-databricks_premerge-github ID:216


Currently databricks runtimes (10.4,11.4) passed the same cases well, but 12.2 runtime (spark332db) would fail
`fixed_length_byte_array.parquet-Exception: Failed: DID NOT RAISE <class 'Exception'>`

```
[2023-09-13T07:56:37.609Z] FAILED ../../src/main/python/parquet_testing_test.py::test_parquet_testing_error_files[confs1-/home/ubuntu/spark-rapids/thirdparty/parquet-testing/data/fixed_length_byte_array.parquet-Exception] - Failed: DID NOT RAISE <class 'Exception'>
[2023-09-13T07:56:37.609Z] FAILED ../../src/main/python/parquet_testing_test.py::test_parquet_testing_error_files[confs0-/home/ubuntu/spark-rapids/thirdparty/parquet-testing/data/fixed_length_byte_array.parquet-Exception][INJECT_OOM] - Failed: DID NOT RAISE <class 'Exception'>
```

```[2023-09-13T05:58:45.805Z] _ test_parquet_testing_error_files[confs0-/home/ubuntu/spark-rapids/thirdparty/parquet-testing/data/fixed_length_byte_array.parquet-Exception] _
[2023-09-13T05:58:45.805Z] [gw3] linux -- Python 3.8.10 /usr/bin/python
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z] path = '/home/ubuntu/spark-rapids/thirdparty/parquet-testing/data/fixed_length_byte_array.parquet'
[2023-09-13T05:58:45.805Z] errobj = 'Exception'
[2023-09-13T05:58:45.805Z] confs = {'spark.rapids.sql.format.parquet.reader.footer.type': 'NATIVE', 'spark.sql.legacy.parquet.datetimeRebaseModeInRead': 'CORRECTED', 'spark.sql.legacy.parquet.int96RebaseModeInRead': 'CORRECTED'}
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z]     @pytest.mark.parametrize(("path", "errobj"), gen_testing_params_for_errors())
[2023-09-13T05:58:45.805Z]     @pytest.mark.parametrize("confs", [_native_reader_confs, _java_reader_confs])
[2023-09-13T05:58:45.805Z]     def test_parquet_testing_error_files(path, errobj, confs):
[2023-09-13T05:58:45.805Z]         error_msg = errobj
[2023-09-13T05:58:45.805Z]         print("error_msg:", error_msg)
[2023-09-13T05:58:45.805Z]         if type(error_msg) != str:
[2023-09-13T05:58:45.805Z]             error_msg, xfail_reason = errobj
[2023-09-13T05:58:45.805Z]             pytest.xfail(xfail_reason)
[2023-09-13T05:58:45.805Z] >       assert_gpu_and_cpu_error(
[2023-09-13T05:58:45.805Z]             lambda spark: spark.read.parquet(path).collect(),
[2023-09-13T05:58:45.805Z]             conf=confs,
[2023-09-13T05:58:45.805Z]             error_message=error_msg)
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z] ../../src/main/python/parquet_testing_test.py:137: 
[2023-09-13T05:58:45.805Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2023-09-13T05:58:45.805Z] ../../src/main/python/asserts.py:632: in assert_gpu_and_cpu_error
[2023-09-13T05:58:45.805Z]     assert_spark_exception(lambda: with_gpu_session(df_fun, conf), error_message)
[2023-09-13T05:58:45.805Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z] func = <function assert_gpu_and_cpu_error.<locals>.<lambda> at 0x7f6010480b80>
[2023-09-13T05:58:45.805Z] error_message = 'Exception'
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z]     def assert_spark_exception(func, error_message):
[2023-09-13T05:58:45.805Z]         """
[2023-09-13T05:58:45.805Z]         Assert that a specific Java exception is thrown
[2023-09-13T05:58:45.805Z]         :param func: a function to be verified
[2023-09-13T05:58:45.805Z]         :param error_message: a string such as the one produce by java.lang.Exception.toString
[2023-09-13T05:58:45.805Z]         :return: Assertion failure if no exception matching error_message has occurred.
[2023-09-13T05:58:45.805Z]         """
[2023-09-13T05:58:45.805Z]         with pytest.raises(Exception) as excinfo:
[2023-09-13T05:58:45.805Z] >           func()
[2023-09-13T05:58:45.805Z] E           Failed: DID NOT RAISE <class 'Exception'>
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z] ../../src/main/python/asserts.py:618: Failed
[2023-09-13T05:58:45.805Z] ----------------------------- Captured stdout call -----------------------------
[2023-09-13T05:58:45.805Z] error_msg: Exception
[2023-09-13T05:58:45.805Z] _ test_parquet_testing_error_files[confs1-/home/ubuntu/spark-rapids/thirdparty/parquet-testing/data/fixed_length_byte_array.parquet-Exception] _
[2023-09-13T05:58:45.805Z] [gw3] linux -- Python 3.8.10 /usr/bin/python
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z] path = '/home/ubuntu/spark-rapids/thirdparty/parquet-testing/data/fixed_length_byte_array.parquet'
[2023-09-13T05:58:45.805Z] errobj = 'Exception'
[2023-09-13T05:58:45.805Z] confs = {'spark.rapids.sql.format.parquet.reader.footer.type': 'JAVA', 'spark.sql.legacy.parquet.datetimeRebaseModeInRead': 'CORRECTED', 'spark.sql.legacy.parquet.int96RebaseModeInRead': 'CORRECTED'}
[2023-09-13T05:58:45.805Z] 
[2023-09-13T05:58:45.805Z]     @pytest.mark.parametrize(("path", "errobj"), gen_testing_params_for_errors())
[2023-09-13T05:58:45.805Z]     @pytest.mark.parametrize("confs", [_native_reader_confs, _java_reader_confs])
[2023-09-13T05:58:45.805Z]     def test_parquet_testing_error_files(path, errobj, confs):
[2023-09-13T05:58:45.805Z]         error_msg = errobj
[2023-09-13T05:58:45.805Z]         print("error_msg:", error_msg)
[2023-09-13T05:58:45.805Z]         if type(error_msg) != str:
[2023-09-13T05:58:45.805Z]             error_msg, xfail_reason = errobj
[2023-09-13T05:58:45.805Z]             pytest.xfail(xfail_reason)
[2023-09-13T05:58:45.805Z] >       assert_gpu_and_cpu_error(
[2023-09-13T05:58:45.806Z]             lambda spark: spark.read.parquet(path).collect(),
[2023-09-13T05:58:45.806Z]             conf=confs,
[2023-09-13T05:58:45.806Z]             error_message=error_msg)
[2023-09-13T05:58:45.806Z] 
[2023-09-13T05:58:45.806Z] ../../src/main/python/parquet_testing_test.py:137: 
[2023-09-13T05:58:45.806Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2023-09-13T05:58:45.806Z] ../../src/main/python/asserts.py:632: in assert_gpu_and_cpu_error
[2023-09-13T05:58:45.806Z]     assert_spark_exception(lambda: with_gpu_session(df_fun, conf), error_message)
[2023-09-13T05:58:45.806Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2023-09-13T05:58:45.806Z] 
[2023-09-13T05:58:45.806Z] func = <function assert_gpu_and_cpu_error.<locals>.<lambda> at 0x7f60022bd310>
[2023-09-13T05:58:45.806Z] error_message = 'Exception'
[2023-09-13T05:58:45.806Z] 
[2023-09-13T05:58:45.806Z]     def assert_spark_exception(func, error_message):
[2023-09-13T05:58:45.806Z]         """
[2023-09-13T05:58:45.806Z]         Assert that a specific Java exception is thrown
[2023-09-13T05:58:45.806Z]         :param func: a function to be verified
[2023-09-13T05:58:45.806Z]         :param error_message: a string such as the one produce by java.lang.Exception.toString
[2023-09-13T05:58:45.806Z]         :return: Assertion failure if no exception matching error_message has occurred.
[2023-09-13T05:58:45.806Z]         """
[2023-09-13T05:58:45.806Z]         with pytest.raises(Exception) as excinfo:
[2023-09-13T05:58:45.806Z] >           func()
[2023-09-13T05:58:45.806Z] E           Failed: DID NOT RAISE <class 'Exception'>
```

**Steps/Code to reproduce bug**
run integration case in databricks 12.2 runtime

**Expected behavior**
Pass the test

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/integration_tests/src/main/python/parquet_testing_test.py
+++ b/integration_tests/src/main/python/parquet_testing_test.py
@@ -39,7 +39,6 @@ _java_reader_confs = copy_and_update(
 # When the association is a pair rather than a string, it's a way to xfail the test
 # by providing the error string and xfail reason.
 _error_files = {
-    "fixed_length_byte_array.parquet": "Exception",
     "large_string_map.brotli.parquet": "Exception",
     "nation.dict-malformed.parquet": ("Exception", "https://github.com/NVIDIA/spark-rapids/issues/8644"),
     "non_hadoop_lz4_compressed.parquet": "Exception",
@@ -50,6 +49,7 @@ _error_files = {
 # xfail reason message.
 _xfail_files = {
     "byte_array_decimal.parquet": "https://github.com/NVIDIA/spark-rapids/issues/8629",
+    "fixed_length_byte_array.parquet": "https://github.com/rapidsai/cudf/issues/14104",
     "datapage_v2.snappy.parquet": "datapage v2 not supported by cudf",
     "delta_binary_packed.parquet": "https://github.com/rapidsai/cudf/issues/13501",
     "delta_byte_array.parquet": "https://github.com/rapidsai/cudf/issues/13501",

